geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+
+###  Gateway Conditions
+
+`Gateway` currently supports the following top-level condition types: `Accepted` and `Programmed`
+
+For a `Gateway`, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a `ListenerSet` listener is invalid, `ListenersNotValid` would be reported. 
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+Parent gateways MUST NOT have `ListenerSet` listeners in their `status.listeners` conditions list.
+
+It is up to the implementation whether an invalid listener affects other listeners in the Gateway.
+
+### ListenerSet Conditions
+
+`ListenerSets` MUST NOT have their parent `Gateway`'s' listeners in the `status.listeners` conditions list.  An implementation MAY reject listeners with `ListenerConditionAccepted=False` and Reason `TooManyListeners` `ListenerSets`, like a `Gateway`, also have two top-level conditions: `Accepted` and `Programmed`. These conditions, when surfacing details about listeners, MUST only summarize the `status.listener` conditions that are exclusive to the `ListenerSet`. 
+
+These conditions MUST also surface top-level `Gateway` conditions that impact the `ListenerSet`. For example, if a `Gateway` requests an invalid address and it cannot be accepted/programmed then the `ListenerSet`'s' `Accepted` condition MUST be set to `False`.
+
+For example, if I have a `Gateway` named `parent`, and two `ListenerSets` named `child-1`, and `child-2` then:
+* If `parent` is entirely invalid (for example, an invalid `address`) and `Accepted=False`, all two `ListenerSets` will reported `Accepted=False`.
+* If `child-1` has an invalid listener, `parent` and `child-1` will report `ListenersNotValid`, while `child-2` will not.
+* If `child-1` references a parent that doesn't allow merging then `child-1` will report `Accepted=False`
+* If `child-1` references another child (eg. `child-2`) then `child-1` will report `Accepted=False` 
+* If `child-1` is valid, then when `child-2` is created if it conflicts with `child-1` then `child-2` will report `Accepted=False`. `child-1` status conditions will remain unchanged. `parent` will report `ListenersNotValid`
+
+When reporting status of a child, an implementation SHOULD be cautious about what information from the parent or siblings are reported
+to avoid accidentally leaking sensitive information that the child would not otherwise have access to.
+
+TBD: The following situations aren't possible to handle since there's no 'gatewayClass' on the `ListenerSet`
+* If `child-1` references a parent that doesn't exist then `child-1` will report `Accepted=False`
+* If `child-1` references itself then `child-1` will report `Accepted=False`
+* If `child-1` and `parent` have different gatewayClassNames then `child-1` will report `Accepted=False`
```

Comment:
This is actually a similar class of problem to what happens if Routes point to a non-existent Gateway - there's no way to be sure that the Route is actually relevant for your implementation, so you really shouldn't update the status.

Reply:
Stepping back a bit, why would the child even be able to specify a GatewayClassName to begin with?

Reply:
I'm going to delete this block - I wanted to capture some failure modes that we cannot handle _unless_ ListenerSet had a gatewayClass param